voorhoede/head-start
# chore: seed docs script

# Changes

- Adds seed script documentation, creating (or updating) CMS records for all `docs/*.md` files.
- Adds a documentation page (if none exists; with slug `documentation`) and uses it as a parent page.
- Adds a page partial (if none exists; titled "Documentation index") as a convenient list of links to all documentation pages.
- Note: dependencies are kept outside root package.json to resolve conflicts. Also the seed dependencies are not needed for other projects using Head Start.

# Associated issue

Related to #27, though that issue is mainly about seeding required content.

# How to test

1. Check out this feature branch
2. Run `npm run seed:docs` (you may want to switch to a new sandbox environment for this)
3. Verify the docs are added to the current DatoCMS environment (in `./datocms-environment.ts`)
4. Update a documentation file
5. Re-run the script
6. Verify the related record in the CMS is updated
7. Review the documentation pages in the browser
8. Verify links in the pages (relative and external) work correctly

# Checklist

- ~I have added a decision log entry if the change affects the architecture or changes a significant technology~
